src/app/ir/checking/findBadConvChannels.ts

```diff
@@ -0,0 +1,36 @@
+import Graph from '@/app/ir/Graph';
+import Conv1d from '@/app/ir/model/conv1d';
+import Conv2d from '@/app/ir/model/conv2d';
+import Conv3d from '@/app/ir/model/conv3d';
+import GraphNode from '@/app/ir/GraphNode';
+import IrError from '@/app/ir/checking/irError';
+import { Severity } from '@/app/ir/checking/severity';
+
+/**
+ * Finds all nodes in the graph that are right before a Conv node and are Conv
+ * nodes themselves, and checks their channels match.
+ */
+export const findBadConvChannels = (graph: Graph) => {
+  type Conv = Conv1d | Conv2d | Conv3d;
```

Comment:
ConvTranspose1d, ConvTranspose2d, ConvTranspose3d can also be shape checked in this way. could add them too?

Reply:
but does this check happen between 2 layers A and B where A is Conv2d and B is ConvTranspose3d?
And vice versa

Reply:
yeah could do. ConvTranspose is kind of like inverse convolution. The in_channels and out_channels check should still be possible between all of them I think

Reply:
_you think_ ahahah sure nws